ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -203,17 +201,16 @@ private[platform] object BufferedTransactionsReader {
 
     val transactionsSource = Timed.source(
       sourceTimer, {
-        // TODO Scala 2.13.5: Remove the @unchecked once migrated to Scala 2.13.5 where this false positive exhaustivity check for Vectors is fixed
-        (transactionsBuffer.slice(startExclusive, endInclusive): @unchecked) match {
+        transactionsBuffer.slice(startExclusive, endInclusive) match {
```

Comment:
@tudor-da tried it with 2.13.8, it complained still for some weird case. Refactored without doing destructuring so we are not messing with exhaustiveness-checks. WDYT?

Reply:
Now it can throw on empty slice, won't it?

Reply:
Hmm, not but only because it's guarded by the above. I don't usually like when we need to rely on ordering in pattern matching.

Reply:
I see...seems to me unapply on +: trips for 1 sized vectors (like :: for Lists), so for the case size == 1 the order mattered originally as well, or am I missing something?
Anyway I have no strong opinion, I can add a guard for the second like `if slice.size > 1` would you like me to do so?

Reply:
You mean the `+:` extractor was unsound? Odd. Anyway, Scala's problem. Yes, that guard is best. And it's guaranteed that it will stare at anyone often enough so it will be addressed once Scala fixes it

Reply:
Fixes it? It only matches non-empty...as with :: not sure I see the problem here :thinking:
Anyway happy to add a guard

Reply:
I think @nmarton-da you tried `+:` instead of `:+` - the latter will allow to pattern match the usual way:
```
          case BufferSlice.Prefix(head +: tail) =>
            fetchTransactions(startExclusive, head._1, filter, verbose)
              .concat(bufferedSource(tail))
              .mapMaterializedValue(_ => NotUsed)

          case BufferSlice.Prefix(_) =>
            fetchTransactions(startExclusive, endInclusive, filter, verbose)

Reply:
hehe guard is not working, not sure why I thought scala type checking can check two boolean expression for exhaustion :man_facepalming: adding an nested if. That should serve your intent I guess